src/categories.cpp

```diff
@@ -38,75 +40,128 @@ QString CategoryFactory::categoriesFilePath()
         int parentID = cells[3].trimmed().toInt(&cell3Ok);
         if (!cell0Ok || !cell3Ok) {
-          log::error("invalid category line {}: {}", lineNum, line.constData());
+          log::error(tr("invalid category line {}: {}").toStdString(), lineNum,
+                     line.constData());
         }
-        addCategory(id, QString::fromUtf8(cells[1].constData()), nexusIDs, parentID);
+        addCategory(id, QString::fromUtf8(cells[1].constData()), nexusCats, parentID);
+      } else if (cells.count() == 3) {
+        bool cell0Ok = true;
+        bool cell3Ok = true;
+        int id       = cells[0].toInt(&cell0Ok);
+        int parentID = cells[2].trimmed().toInt(&cell3Ok);
+        if (!cell0Ok || !cell3Ok) {
+          log::error(tr("invalid category line {}: {}").toStdString(), lineNum,
+                     line.constData());
+        }
+
+        addCategory(id, QString::fromUtf8(cells[1].constData()),
+                    std::vector<NexusCategory>(), parentID);
+      } else {
+        log::error(tr("invalid category line {}: {} ({} cells)").toStdString(), lineNum,
+                   line.constData(), cells.count());
       }
     }
     categoryFile.close();
+
+    QFile nexusMapFile(nexusMappingFilePath());
+    if (!nexusMapFile.open(QIODevice::ReadOnly)) {
+      needLoad = true;
+    } else {
+      int nexLineNum = 0;
+      while (!nexusMapFile.atEnd()) {
+        QByteArray nexLine = nexusMapFile.readLine();
+        ++nexLineNum;
+        QList<QByteArray> nexCells = nexLine.split('|');
+        if (nexCells.count() == 3) {
+          std::vector<NexusCategory> nexusCats;
+          QString nexName = nexCells[1];
+          bool ok         = false;
+          int nexID       = nexCells[2].toInt(&ok);
+          if (!ok) {
+            log::error(tr("invalid nexus ID {}").toStdString(),
+                       nexCells[2].constData());
+          }
+          int catID = nexCells[0].toInt(&ok);
+          if (!ok) {
+            log::error(tr("invalid category id {}").toStdString(),
+                       nexCells[0].constData());
+          }
+          m_NexusMap.insert_or_assign(nexID, NexusCategory(nexName, nexID));
+          m_NexusMap.at(nexID).m_CategoryID = catID;
+        } else {
+          log::error(tr("invalid nexus category line {}: {} ({} cells)").toStdString(),
+                     lineNum, nexLine.constData(), nexCells.count());
+        }
+      }
+    }
+    nexusMapFile.close();
   }
   std::sort(m_Categories.begin(), m_Categories.end());
   setParents();
+  if (needLoad)
+    loadDefaultCategories();
 }
 
-CategoryFactory& CategoryFactory::instance()
+CategoryFactory* CategoryFactory::instance()
 {
-  if (s_Instance == nullptr) {
-    s_Instance = new CategoryFactory;
-  }
-  return *s_Instance;
+  static CategoryFactory s_Instance;
+  return &s_Instance;
 }
 
 void CategoryFactory::reset()
 {
   m_Categories.clear();
+  m_NexusMap.clear();
   m_IDMap.clear();
   // 28 =
```

Comment:
Maybe remove the comments since the value are not hardcoded anymore.

Reply:
I have a feeling these were somehow left behind after a rebase at some point.